featuretools/synthesis/dfs.py

```diff
@@ -261,6 +262,10 @@ def dfs(dataframes=None,
     features = dfs_object.build_features(
         verbose=verbose, return_types=return_types)
 
+    assert (isinstance(features, list) and features != [] and
+            all([isinstance(feature, FeatureBase) for feature in features])), \
```

Comment:
let's limit this to checking for an empty list

Reply:
Will to move the checking after feature_only process resolve this?

Reply:
No, I'd also like to simplify the assert -- I'd like to tie this specific error message to the feature list being empty.  The other two situations will get caught by the check in `calculate_feature_matrix` , and would suggest some pretty strange things going on that might not fit this new error message

Reply:
updated to assert only with features != []